src/lib.rs

```diff
@@ -1,91 +1,108 @@
-//! data it contains.
+//! This crate provides an easy-to-use client for [TiKV](https://github.com/tikv/tikv), a
+//! distributed, transactional key-value database written in Rust.
+//!
+//! This crate lets you connect to a TiKV cluster and use either a transactional or raw (simple
+//! get/put style without transactional consistency guarantees) API to access and update your data.
+//!
+//! The TiKV Rust client supports several levels of abstraction. The most convenient way to use the
+//! client is via [`RawClient`] and [`TransactionClient`]. This gives a very high-level API which
+//! mostly abstracts over the distributed nature of the store and has sensible defaults for all
+//! protocols. This interface can be configured, primarily when creating the client or transaction
+//! objects via the [`Config`] and [`TransactionOptions`] structs. Using some options, you can take
+//! over parts of the protocols (such as retrying failed messages) yourself.
+//!
+//! The lowest level of abstraction is to create and send gRPC messages directly to TiKV (and PD)
+//! nodes. The `tikv-client-store` and `tikv-client-pd` crates make this easier than using the
+//! protobuf definitions and a gRPC library directly, but give you the same level of control.
+//!
+//! In between these levels of abstraction, you can send and receive individual messages to the TiKV
+//! cluster, but take advantage of library code for common operations such as resolving data to
+//! regions and thus nodes in the cluster, or retrying failed messages. This can be useful for
+//! testing a TiKV cluster or for some advanced use cases. See the [`request`] module for
+//! this API, and [`raw::lowering`] and [`transaction::lowering`] for
+//! convenience methods for creating request objects.
 //!
 //! ## Choosing an API
 //!
-//! This crate offers both [**raw**](raw/index.html) and
-//! [**transactional**](transaction/index.html) APIs. You should choose just one for your system.
+//! This crate offers both [raw](RawClient) and
+//! [transactional](Transaction) APIs. You should choose just one for your system.
 //!
-//! The *consequence* of supporting transactions is increased overhead of coordination with the
-//! placement driver for timestamp acquisition. This is approximately 1 RTT.
+//! The consequence of supporting transactions is increased overhead of coordination with the
+//! placement driver and TiKV, and additional code complexity.
 //!
 //! *While it is possible to use both APIs at the same time, doing so is unsafe and unsupported.*
 //!
-//! Choose the one that suites your needs as described below, then add the import statement to your
-//! file where you need to use the library.
-//!
 //! ### Transactional
 //!
-//! The [transactional](transaction/index.html) API supports **transactions** via Multi-Version
-//! Concurrency Control (MVCC).
+//! The [transactional](Transaction) API supports **transactions** via multi-version
+//! concurrency control (MVCC).
```

Comment:
Shall we put a note of garbage collection here?

Reply:
Why do you think we should?

Reply:
Because new users may not be aware of the garbage issue?